examples/nextjs-app-router/components/Form.tsx

```diff
@@ -17,15 +17,29 @@ import {
 	validateForm,
 } from "simple-stack-form/module";
 import { twMerge } from "tailwind-merge";
+import { useFormState as useForStateBase } from "react-dom";
+
+type ValidateFormResult = Awaited<ReturnType<typeof validateForm>> | null;
+
+export function useFormState(
+	action: (formData: FormData) => Promise<ValidateFormResult>,
+) {
+	return useForStateBase(
+		async (_: ValidateFormResult, formData: FormData) => action(formData),
+		null,
+	);
+}
 
 export function useCreateFormContext(
 	validator: FormValidator,
 	fieldErrors?: FieldErrors,
+	submitted?: boolean,
```

Comment:
We can revert any changes to this file that pipe `submitted` through. We'll revisit as a separate PR to decouple `submitStatus` and let users bring their own state tracking